src/Controller/EnMarche/UserController.php

```diff
@@ -26,13 +28,29 @@ class UserController extends Controller
 
     /**
      * @Route("", name="app_user_profile")
-     * @Method("GET|POST")
+     * @Method("GET")
      */
     public function profileOverviewAction(): Response
     {
         return $this->render('user/my_account.html.twig');
     }
 
+    /**
+     * @Route("/mes-dons", name="app_user_profile_donation")
+     * @Method("GET")
+     */
+    public function profileDonationAction(): Response
+    {
+        $donationRepos = $this->getDoctrine()->getRepository(Donation::class);
+        /** @var Adherent $user */
+        $user = $this->getUser();
+
+        return $this->render('user/my_donation.html.twig', [
+            'donations' => $donationRepos->findBy(['emailAddress' => $user->getEmailAddress()]),
+            'last_donation' => $donationRepos->findLastDonation($user->getEmailAddress()),
```

Comment:
to prevent an additional query: can't you retrieve this `last_donation` from the `donations` above ?
by ordering it and getting the first or last element maybe?

Reply:
Yes ;)